mlplumber/steps.py

```diff
@@ -1,22 +1,23 @@
-from baikal import Step
+from baikal import make_step
 from sklearn_pandas import DataFrameMapper
 
 
-class DataFrameMapperStep(Step, DataFrameMapper):
-
-    def __init__(self, *args, name=None, n_outputs=1, **kwargs):
-        super().__init__(*args, name=name, n_outputs=n_outputs, **kwargs)
-        self._nodes = []
-
-    def __getstate__(self):
-        state = super().__getstate__()
-        state["_name"] = self._name
-        state["_nodes"] = self._nodes
-        state["_n_outputs"] = self._n_outputs
-        return state
-
-    def __setstate__(self, state):
-        self._name = state["_name"]
-        self._nodes = state["_nodes"]
-        self._n_outputs = state["_n_outputs"]
-        super().__setstate__(state)
```

Comment:
Since the base `DataFrameMapper` class defines its own `__setstate__` (I'm not sure why it does not define `__getstate__`, though) you still need to define these functions if you want the step to be pickle-able (same situation as when dealing with CatBoost estimators).

You may want to create an intermediate `_PickleableStep` class that inherits from `Step` and define these methods.

```python
class _PickleableStep(Step):
    def __getstate__(self):
        state = super().__getstate__()
        state["_name"] = self._name
        state["_nodes"] = self._nodes
        state["_n_outputs"] = self._n_outputs
        return state

    def __setstate__(self, state):
        self._name = state.pop("_name")
        self._nodes = state.pop("_nodes")
        self._n_outputs = state.pop("_n_outputs")
        super().__setstate__(state)

class DataFrameMapperStep(_PickleableStep, DataFrameMapper):
    ...
```

Note that you don't need that `self._nodes = []` in `__init__`.

Reply:
(At first glance one would think that baikal could include these methods, but that actually would be create more problems than solve them. Things are easier if `__getstate__` and `__setstate__` are defined at the end of the class hierarchy or, even better (though unfortunately less realistically), if they are not overriden at all.)